File: apple/Sources/ReactNativeWatchOS/Modifiers/GlassEffectTransition.swift
import SwiftUI

/// SwiftUI `.glassEffectTransition(_:)` — watchOS 26 "Liquid Glass".
/// Selects how a glass element animates as it's inserted or removed.
///
/// `transition`: `materialize` (default Liquid Glass materialize),
/// `identity` (no transition), or `matchedGeometry` (driven from a
/// matched-geometry source).
///
/// Gated to watchOS 26+ — returns the view unchanged on older OS.
enum RNWGlassEffectTransitionModifier {
    @MainActor
    static func register(into r: RNWModifierRegistry) {
        r.register("glassEffectTransition") { view, params, _ in
            guard #available(watchOS 26.0, *) else { return view }

            let transition: GlassEffectTransition
            switch params.string("transition") {
            case "identity":
                transition = .identity
            case "matchedGeometry":
                transition = .matchedGeometry
            default:
                transition = .materialize
            }
            return AnyView(view.glassEffectTransition(transition))
        }
    }
}

File: src/modifiers/glassEffectTransition.ts
import { createModifier } from './createModifier';

/// How a glass element animates in/out as it appears or disappears.
/// `materialize` is the default Liquid Glass materialize transition;
/// `identity` disables the transition; `matchedGeometry` drives the
/// transition from a matched-geometry source.
export type GlassEffectTransitionType =
  | 'materialize'
  | 'identity'
  | 'matchedGeometry';

export interface GlassEffectTransitionParams {
  /// The transition style. Defaults to `materialize`.
  transition?: GlassEffectTransitionType;
}

/// SwiftUI `.glassEffectTransition(_:)` (watchOS 26 "Liquid Glass").
/// Controls how the glass effect transitions when the view is inserted or
/// removed. No-op on watchOS < 26.
export function glassEffectTransition(
  transition: GlassEffectTransitionType = 'materialize'
) {
  return createModifier('glassEffectTransition', { transition });
}
